modules/video/src/ecc.cpp

```diff
@@ -323,17 +323,34 @@ double cv::computeECC(InputArray templateImage, InputArray inputImage, InputArra
     Scalar meanTemplate, sdTemplate;
 
     int active_pixels = inputMask.empty() ? templateImage.size().area() : countNonZero(inputMask);
-
+    int type = templateImage.type();
     meanStdDev(templateImage, meanTemplate, sdTemplate, inputMask);
     Mat templateImage_zeromean = Mat::zeros(templateImage.size(), templateImage.type());
-    subtract(templateImage, meanTemplate, templateImage_zeromean, inputMask);
+    Mat templateMat = templateImage.getMat();
+    Mat inputMat = inputImage.getMat();
+
+    /*
+     * For unsigned ints, when the mean is computed and subtracted, any values less than the mean
+     * will be set to 0 (since there are no negatives values). This impacts the norm and dot product, which
+     * ultimately results in an incorrect ECC. To circumvent this problem, if unsigned ints are provided,
+     * we convert them to a signed ints with larger resolution for the subtraction step.
+     */
+    if(type == CV_8U) {
+        templateMat.convertTo(templateMat, CV_16S);
+        inputMat.convertTo(inputMat, CV_16S);
```

Comment:
Please avoid inplace src / dst parameters.
Current implementation is ineffective (extra copy can be done due to non-inplace operation).

Use swap() to update "local" Mat.

Reply:
Done - thank you!